generation/src/proof_generation/k/kore_convertion/language_semantics.py

```diff
@@ -692,9 +688,7 @@ def _convert_pattern(self, scope: ConvertionScope, pattern: kore.Pattern) -> Pat
                 exists_outer_sort_pattern: Pattern = self._convert_sort(scope, sort)
                 exists_var_pattern: Pattern = self._convert_pattern(scope, var)
                 exists_pattern: Pattern = self._convert_pattern(scope, pattern)
-                # TODO: This should be fixed after functional substitution is implemented
-                # assert isinstance(exists_var_pattern, EVar | SVar)
-                assert isinstance(exists_var_pattern, MetaVar)
+                assert isinstance(exists_var_pattern, EVar | SVar)
```

Comment:
```suggestion
                assert isinstance(exists_var_pattern, EVar)
```

there is no existential binder for set variables in matching logic

Reply:
You're right. I just went by the comment.